lib/BloqadeExpr/src/types.jl

```diff
@@ -281,6 +281,51 @@ end
 
 SumOfZ(n::Int) = SumOfZ(n, 1)
 
+
+# this subtype is a bit fishy here
+# There are some places where this 
+# helps overload some functions but 
+# for clarity it might make sense to 
+# to make this a unique type.
+struct RydbergHamiltonian <: AbstractTerm
+    Terms::AbstractBlock
```

Comment:
I think it's better to just have three terms separated instead of this. It's easier for checking. Then you don't need to have the field C. 


Also note we don't have a style guide but we roughly follow the YASGuide: https://github.com/jrevels/YASGuide

Here the field name should be in snakecase.

Reply:
Without the field `C` the package fails [this](https://github.com/QuEraComputing/Bloqade.jl/blob/32c07c7cdc02b92f59ee9e661c576c97b7cc08f6/lib/BloqadeExpr/test/units.jl#L16) units test. This isn't a big deal but I'm assuming having the field `C` accessible is necessary for other parts of Bloqade?

Reply:
I think we can change that test for this case. We haven't made internal fields as a stable API so I think this is fine. Plus C is kinda considered a hardware detail in general so I think not much places is gonna use that.

Reply:
Alrighty, I will update the test to reflect this change.